src/renderers/dom/shared/eventPlugins/ChangeEventPlugin.js

```diff
@@ -260,34 +143,33 @@ function handleControlledInputBlur(inst, node) {
 var ChangeEventPlugin = {
   eventTypes: eventTypes,
 
-  _isInputEventSupported: isInputEventSupported,
-
-  extractEvents: function(
-    topLevelType,
-    targetInst,
-    nativeEvent,
-    nativeEventTarget,
-  ) {
+  extractEvents(topLevelType, targetInst, nativeEvent, nativeEventTarget) {
     var targetNode = targetInst
       ? ReactDOMComponentTree.getNodeFromInstance(targetInst)
       : window;
 
+    // On the selectionchange event, the target is the document which isn't
+    // helpful becasue we need the input, so we use the activeElement instead.
+    if (!isTextInputEventSupported && topLevelType === 'topSelectionChange') {
+      targetNode = getActiveElement();
+
+      if (targetNode) {
+        targetInst = ReactDOMComponentTree.getInstanceFromNode(targetNode);
+      }
+    }
+
     var getTargetInstFunc, handleEventFunc;
+
     if (shouldUseChangeEvent(targetNode)) {
       getTargetInstFunc = getTargetInstForChangeEvent;
-    } else if (isTextInputElement(targetNode)) {
-      if (isInputEventSupported) {
-        getTargetInstFunc = getTargetInstForInputOrChangeEvent;
-      } else {
-        getTargetInstFunc = getTargetInstForInputEventPolyfill;
-        handleEventFunc = handleEventsForInputEventPolyfill;
-      }
-    } else if (shouldUseClickEvent(targetNode)) {
-      getTargetInstFunc = getTargetInstForClickEvent;
+    } else if (isTextInputElement(targetNode) && !isTextInputEventSupported) {
+      getTargetInstFunc = getTargetInstForInputEventPolyfill;
+    } else {
+      getTargetInstFunc = getTargetInstForInputOrChangeEvent;
     }
 
     if (getTargetInstFunc) {
-      var inst = getTargetInstFunc(topLevelType, targetInst);
+      var inst = getTargetInstFunc(topLevelType, targetInst, targetNode);
       if (inst) {
         var event = createAndAccumulateChangeEvent(
```

Comment:
@jquense can we use `targetNode` as `nativeEventTarget` for selection events in IE9? When you start backspacing it registers a change event but it gets `undefined` for `event.target.value` since `event.target` is still the document.